### PR TITLE
Add doc build to travis

### PR DESCRIPTION
**Pull request by bashtage**
Add a doc build that is currently only for information, it does not raise
Also contains many small doc fixes that came up when creating doc build

### CODE REVIEW

File: docs/source/dev/git_notes.rst [hunk 1]
```diff
@@ -45,22 +45,27 @@ This should take you to your fork's page.
 
 Then, you want to clone the fork to your machine::
 
-    git clone git@github.com:your-user-name/statsmodels.git statsmodels-yourname
-    cd statsmodels-yourname
-    git remote add upstream git://github.com/statsmodels/statsmodels.git
+    git clone https://github.com/your-user-name/statsmodels
```
Comment (josef-pkt) [inline docs/source/dev/git_notes.rst]: why do you switch to the https interface? 
I'm using almost only git@github.com

**Comment by bashtage** [comment]
https is how github presents it self and is less scary for people who need help with git.

File: docs/source/dev/git_notes.rst (hunk 1 above)
Comment (bashtage) [inline docs/source/dev/git_notes.rst]: See also 

https://help.github.com/articles/which-remote-url-should-i-use/

for some reasons why https could be preferred, at least for people starting.

**Comment by bashtage** [comment]
@josef-pkt This one is useful for making it easier to release.  In particular, it adds a doc build where doc errors can be easily seen.  It also starts formally testing the notebooks to make sure that they aren't broken by code changes (ex 3 notebooks that are broken where the fix is beyond this PR -- 2 using predict and 1 with nan leading to a LinearAlgebraError).

All other fixes are doc fixes, which are also good to get in .

**Comment by josef-pkt** [comment]
There are many changes in here that I don't understand. I try to go over it in the next days and ask questions.

main one on top: this reverts #2739 which was a big improvement at the time.
some are changes in docstring that I don't know why they are needed

I don't know much about the changes in `tools/nbgenerate.py` because I leave that up to you and @TomAugspurger

**Comment by bashtage** [comment]
The docker commands never did anything.   IN particular, if you look at the logs you will see

```
docker: unrecognized service
```

since docker is non-trivial on Travis.  

The big win was using conda with is a "many-linux" build so there isn't much dependence on the underlying OS.

**Comment by bashtage** [comment]
The docstring changes are mostly since these were being interpreted as missing cross references.  They also don't match the style.

**Comment by josef-pkt** [comment]
I'm merging this, there might be some smaller questions, but all is green.

Thanks @bashtage

**Pull request merged**